### DISCUSSION TITLE
learn/jep-408-with-jbang/

### DISCUSSION THREAD

**Discussion by giscus** [jbangdev/jbang#1078] [Blog/Website]
# learn/jep-408-with-jbang/

In March JEP: 408: Simple Web Server was announced. One of JEP 408’s goals is to make an easy to use web server available with Java using java -m jdk.httpserver.

https://www.jbang.dev/learn/jep-408-with-jbang/

**Comment by ppalaga** [comment]
This is pure awesomeness!

**Comment by ice09** [comment]
Really nice, however for a really simple intranet with a Raspi (4), it does work reproducibly with `jhttpd -b 0.0.0.0` but not with just `jhttpd`. The log statement is the same for both and if this is the statement executed, I can't see the error: https://github.com/jbangdev/jbang-catalog/blob/89f670b879d41ecf5d1fb8675af7f9413ed4465e/httpd.java#L89

**Reply by ice09** [reply]
sorry, it's the other way around ofc, so just `jhttpd` does not work, even though 0.0.0.0 is displayed as binding, `jhttpd -b 0.0.0.0` works. I can reproduce this every time.

**Reply by ice09** [reply]
To add, locally on the raspi it works, just not from another computer in the intranet. With `-b 0.0.0.0` it works from all computers in the intranet. Not working means no connect (connection refused).

**Reply by quintesse** [reply]
I think the problem is that the default value "0.0.0.0" is only set for the logger output, it's _not_ used for actually setting the interface for the server! https://github.com/jbangdev/jbang-catalog/blob/89f670b879d41ecf5d1fb8675af7f9413ed4465e/httpd.java#L83

**Reply by quintesse** [reply]
The docs for `getByName()` say "If the host is null then an InetAddress representing an address of the loopback interface is returned.". I think the easiest fix would be to just set a default value for `bind` and remove the `bindLabel` line. Now if the default should be 0.0.0.0 or 127.0.0.1 is another discussion. (I'd say 127.0.0.1 because it's a bit more security-conscious, it would require people to explicitly set it to 0.0.0.0 to enable external access).

**Reply by quintesse** [reply]
Fixed. (You might need to run once with `jbang --fresh httpd@jbangdev` or `jbang cache clear` to make Jbang use the latest version of the code)

**Reply by ice09** [reply]
Haha, great catch, didn't see that it was used for logging only. Thx, it works now (ie. displays the correct 127.0.0.1 if none given).
